packages/base/src/3dview/mainview.tsx

```diff
@@ -612,22 +612,35 @@ export class MainView extends React.Component<IProps, IStates> {
         if (meshGroup.visible) {
           this._boundingGroup.expandByObject(mainMesh);
         }
+
+        // Save original color for the main mesh
+        if (mainMesh.material?.color) {
+          const originalMeshColor = new THREE.Color(
+            objColor || DEFAULT_MESH_COLOR
+          );
+          if (!selected) {
+            mainMesh.material.color = originalMeshColor;
+          }
+          mainMesh.userData.originalColor = originalMeshColor.clone();
+        }
+
         if (selected) {
           this._selectedMeshes.push(mainMesh);
         }
         edgesMeshes.forEach(el => {
           this._edgeMaterials.push(el.material);
+          const originalEdgeColor = new THREE.Color(objColor).multiplyScalar(
+            0.7
+          );
           if (selectedNames.includes(el.name)) {
             this._selectedMeshes.push(el as any as BasicMesh);
             el.material.color = SELECTED_MESH_COLOR;
             el.material.linewidth = SELECTED_LINEWIDTH;
-            el.userData.originalColor = new THREE.Color(objColor);
+            el.userData.originalColor = originalEdgeColor.clone();
```

Comment:
I believe that `new THREE.Color(originalEdgeColor)` & `originalEdgeColor.clone()` serves the same puspose, hence I felt like using this instead of again defining the color.